conjure-java-core/src/test/resources/dangerous-name-service.yml

```diff
@@ -0,0 +1,33 @@
+services:
+  NameCollisionService:
+    name: Naming Collision Service
+    package: com.palantir.product
+    base-path: /
+
+    endpoints:
+      int:
+        http: GET /
+        auth: header
+        args:
+          authHeader:
+            param-type: query
+            type: string
+          long:
+            param-type: query
+            type: string
+          runtime:
+            param-type: query
+            type: string
+          serializer:
+            param-type: query
+            type: string
+          deserializer:
+            param-type: query
+            type: string
+          delegate:
+            param-type: query
+            type: string
+          result:
+            param-type: query
+            type: string
```

Comment:
what happen if you declare params of the name authHeader_, long_, runtime_, serializer_, ... (or are those not valid conjure names)

Reply:
`Parameter names in endpoint paths and service definitions must match pattern ^[a-z][a-z0-9]*([A-Z0-9][a-z0-9]+)*$`

The same isn't true for method names, so `int` is renamed to `int_`, which may conflict with another defined endpoint `int_`. That case is a little bit _too_ weird, so I'm fine letting that one fail for now. We'll only automatically fix so many bad things for you :-)

Reply:
Is the parameter names pattern actually enforced?
A possible fix for method names is to always add "_" if the name end with "_".

Reply:
Parameter name regex is enforced when the conjure schema file is parsed.I don't follow your solution for method names, the problem is if I have a method named `int` as well as `int_`. the `int_` method is fine, but attempting to escape `int` produces a method name which matches the existing `int_` method.

Reply:
@carterkozak Yes, that's why if in addition to escape `int` to `int_` you also escape `int_` to `int__` it works. To avoid collision you need the function to be bijective, which mean you need to also displace every element of the codomain

Reply:
(I edited the n-2 comment which was weirdly formatted so hard to understand)

Reply:
I'd rather avoid renaming things that aren't themselves invalid, my preference is to fail if we cannot provide a simple fix.

Reply:
Isn't my solution a simple fix :) ? Also those are not more invalid than then one that collide with the reserved keywords.

Reply:
There are existing services with methods named things like `boolean_` which would be modified.

Reply:
Right now is probably the ideal time to break, after it will be too late. But fair enough if you think the cost is too high.